### PR TITLE
3D Annotation for Multi-object Object Segmentation

### PR DESCRIPTION
**Pull request by wkentaro**
Close https://github.com/start-jsk/jsk_apc/issues/2172

- [x] Dataset to ROS.
- [x] Annotation tool using interactive marker.
- ~~Do annotation.~~

## Dataset to ROS

publish_dataset_v3

### CODE REVIEW

**Comment by wkentaro** [comment]
## 3D annotation

The problem is inconsistent location of point cloud possibly caused by

- extrinsic parameters mis-calibration https://github.com/start-jsk/jsk_apc/issues/2204
- baxter joint mis-calibration https://github.com/start-jsk/jsk_apc/issues/2219

arc2017_3d_annotation

**Comment by wkentaro** [comment]
Postponing this with https://github.com/start-jsk/jsk_apc/issues/2172

**Comment by wkentaro** [comment]
Postpone this with https://github.com/start-jsk/jsk_apc/issues/2172#issuecomment-315447144

File: jsk_arc2017_common/node_scripts/publish_dataset_v3.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import yaml
+
+import cv_bridge
+import dynamic_reconfigure.server
+from geometry_msgs.msg import TransformStamped
+import genpy.message
+import rospy
+from sensor_msgs.msg import CameraInfo
+from sensor_msgs.msg import Image
+from std_msgs.msg import Header
```
Comment (YutoUchimi) [inline jsk_arc2017_common/node_scripts/publish_dataset_v3.py]: `Header ` imported but unused.

File: jsk_arc2017_common/scripts/create_scenes_dataset_v3.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        tf_camera_from_base = yaml.load(
+            open(osp.join(data_dir, 'tf_camera_rgb_from_base.yaml')))
+
+        return frame_idx, img, depth, camera_info, tf_camera_from_base
+
+
+def main():
+    dataset = DatasetCollectedOnShelfMultiView()
+
+    out_dir = '/data/projects/arc2017/datasets/JSKV3_scenes'
```
Comment (YutoUchimi) [inline jsk_arc2017_common/scripts/create_scenes_dataset_v3.py]: Same as `publish_dataset_v3.py`.

File: jsk_arc2017_common/scripts/create_scenes_dataset_v3.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import yaml
+
+import cv_bridge
+from geometry_msgs.msg import TransformStamped
+import genpy.message
+import rospy
+from sensor_msgs.msg import CameraInfo
+from sensor_msgs.msg import Image
+from std_msgs.msg import Header
+import tf
```
Comment (YutoUchimi) [inline jsk_arc2017_common/scripts/create_scenes_dataset_v3.py]: `cv_bridge`, ..., `tf` imported but unused.

File: jsk_arc2017_common/node_scripts/publish_dataset_v3.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import tf
+
+from jsk_arc2017_common.cfg import PublishDatasetV3Config
+
+
+class DatasetCollectedOnShelfMultiViewScenes(object):
+
+    def __init__(self):
+        self.scene_ids = []
+        self.root = '/data/projects/arc2017/datasets/JSKV3_scenes'
```
Comment (YutoUchimi) [inline jsk_arc2017_common/node_scripts/publish_dataset_v3.py]: The dataset directory should be defined as this? (NOT `/data/...`)
```
import rospkg
PKG_DIR = rospkg.RosPack().get_path('jsk_arc2017_common')
self.root = osp.join(PKG_DIR, 'data/datasets/JSKV3_scenes')
```

Also, please add dataset installation code to `install_data.py` or something like that.

**Comment by wkentaro** [comment]
## Usage

```bash
rosrun jsk_arc2017_common create_scenes_dataset_v3.py
roslaunch jsk_arc2017_common object_segmentation_3d.launch
```

**Pull request merged**